### ISSUE TITLE
[feature-request] Support Extraction to superclass

### ISSUE THREAD

**Issue by Stargator** [dart-lang/sdk#36199]
Thank you for taking the time to file an issue!

In order to route, prioritize, and act on this, please include:

* Dart SDK Version: 2.2.0

----

I think it would be helpful if developers could extract a method or variable to the super class or parent class. Assuming the class is not part of the Dart SDK.

I see it like this:

```dart
class polygon {
// lots of methods/variables 😄 
}

class square extends polygon {
    calculateArea(int sideLength) => sqrt(sideLength);
}
```

After extracting the method, calculateArea() now resides in the polygon class. Thinking about this, it would be a good idea to provide options for making the extracted method an abstract or implemented method depending on use cases.

**Comment by alexander-doroshko** [comment]
@vsmenon, I don't think `area-intellij` is the right label here. Refactorings are first implemented on the Analysis Server end and only then can be adapted on the IDE end.

**Comment by alexander-doroshko** [comment]
Just for the record and as a source for ideas. Java support in IntelliJ IDEA has more general refactoring called [Pull Members Up](https://www.jetbrains.com/help/idea/pull-members-up.html).

**Comment by Stargator** [comment]
@alexander-doroshko Yes! I'm just trying to go through what I like in refactoring efforts, but I am not versed in everything that's possible or already implemented for other languages.

**Comment by escamoteur** [comment]
I really think this refectory should get a little bit more priority. Not having this refectory leads to that many people always define superclasses/interfaces even if they don't need them at the moment, which complicates code unnecessary.
